src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class TaskList {
+    final static BadCommandException INDEX_OOB_ERROR =
+            new BadCommandException("Index given is out of bounds!");
+    protected ArrayList<Task> tasks;
+
+    /**
+     * Initialises an empty TaskList.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+    /**
+     * Initialises an TaskList containing some Tasks.
+     */
+    public TaskList(Collection<Task> initTasks) {
+        tasks = new ArrayList<>(initTasks);
+    }
+
+    private boolean isIndexInRange(int idx) {
+        return idx < tasks.size() && idx >= 0;
+    }
```

Comment:
Seems like you missed out a JavaDoc here. There are some other instances as well.